api/server/router/image/backend.go

```diff
@@ -22,7 +22,7 @@ type Backend interface {
 type imageBackend interface {
 	ImageDelete(imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error)
 	ImageHistory(imageName string) ([]*image.HistoryResponseItem, error)
-	Images(imageFilters filters.Args, all bool, withExtraAttrs bool) ([]*types.ImageSummary, error)
+	Images(opts types.ImageListOptions, withContainerCount bool) ([]*types.ImageSummary, error)
```

Comment:
Curious; was there a reason keeping `withContainerCount` separate, and not add it to the `types.ImageListOptions` ?

Reply:
On a side-note; as we're changing the interface here, wondering if we should take the opportunity to also add a `ctx context.Context` now. (could be a placeholder for now)

Reply:
> Curious; was there a reason keeping `withContainerCount` separate, and not add it to the `types.ImageListOptions` ?

Since this parameter is not exposed via the HTTP API and only used internally, I felt that it is wrong to add it to `api/types`.
In the first iteration I did add it in the `types.ImageListOptions` though. Shall I bring it back in?

Reply:
Yes, I think it makes the interface cleaner, but looks like I should've commented this on https://github.com/moby/moby/pull/42531 (which made the initial change 😅)

Reply:
No problem, added https://github.com/moby/moby/pull/42531/commits/dd61261e5354d97c39bf3be6224c3dcfe6441430